src/injectScripts.js

```diff
@@ -41,8 +46,59 @@ function changeSrcPlugin() {
       attrs: {src: srcPath.replace(TEST_HARNESS, TEST_HARNESS_REPORT)},
     };
 
+    // Add testharnessreport.js src after testharness.js src
     addNode(tree, newSrcNode, (node) => {
-      return node.tag == SCRIPT && node.attrs && node.attrs.src == srcPath;
+      return node.tag === SCRIPT && node.attrs && node.attrs.src === srcPath;
+    });
+  };
+}
+
+// insertTitlePlugin will add <title>description</title> in the following
+// order of precedence if present:
+// will not add if a <title> exists already,
+// immediately after the <head> tag,
+// immediately after the <html> tag,
+// immediately after the <!DOCTYPE html> tag,
+// at the beginning of the document.
+// description is a string.
+function insertTitlePlugin(description) {
+  return function(tree) {
+    let foundTitle = false;
+    tree.match({tag: TITLE}, (node) => {
+      foundTitle = true;
+      return node;
+    });
+    // <title> exists, don't add a new one.
+    if (foundTitle) {
+      console.log('WARNING: existing <title>, not using description parameter');
+      return;
+    }
+    const newTitleNode = {
+      tag: TITLE,
+      content: [description],
+    };
+
+    // Within <head> tag
+    if (addNodeWithinTag(tree, newTitleNode, (node) => {
+      return node.tag === HEAD;
+    })) {
+      return;
+    }
+    // Within <html> tag
+    if (addNodeWithinTag(tree, newTitleNode, (node) => {
+      return node.tag === HTML;
+    })) {
+      return;
+    }
+    // Within <!DOCTYPE html> tag
+    if (addNode(tree, newTitleNode, (node) => {
+      return typeof node === 'string' && node.includes(DOCTYPE);
+    })) {
+      return;
+    }
+    // Default: add title to beginning of document
+    addNode(tree, newTitleNode, () => {
+      return true;
     });
```

Comment:
```suggestion
    addNode(tree, newTitleNode, () => true);
```

Similar for other lambdas above.